packages/graphql/src/translate/create-projection-and-params.ts

```diff
@@ -338,8 +338,7 @@ function createProjectionAndParams({
                 if (optionsInput) {
                     offsetLimitStr = createOffsetLimitStr({ offset: optionsInput.offset, limit: optionsInput.limit });
                 }
-
-                res.projection.push(`${f.alias}: collect(${f.alias})${offsetLimitStr}`);
+                res.projection.push(`${f.alias}: ${!isArray ? "head(" : ""}collect(${f.alias})${offsetLimitStr}${!isArray ? ")" : ""}`);
```

Comment:
Really need to be seeing an integration test or 2 to ensure that this fix works.